deploy/servctl_utils/yaml_settings_utils.py

```diff
@@ -0,0 +1,107 @@
+import collections
+import io
+import jinja2
+import logging
+import os
+import socket
+import time
+import yaml
+
+
+logger = logging.getLogger()
+
+
+def _get_ip_address():
+    """Returns the localhost ip address as a string (eg. "192.168.0.6")."""
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.connect(("8.8.8.8", 80))
+    return s.getsockname()[0]
+
+
+def _parse_jinja_template(jinja_template_path, template_variables):
+    with io.open(jinja_template_path, encoding="UTF-8") as jinja_template_file:
+        try:
+            jinja_template_contents = jinja_template_file.read()
+            yaml_string = jinja2.Template(jinja_template_contents).render(template_variables)
+        except TypeError as e:
+            raise ValueError('unable to render file: %(e)s' % locals())
+
+    return yaml_string
+
+
+def load_settings(settings_file_paths, settings=None):
+    """Parses yaml settings file(s) and returns a dictionary of settings.
+    These yaml files are treated as Jinja templates.
+    If a settings dictionary is also provided as an argument, it will be used as context for Jinja template processing.
+
+    Args:
+        settings_file_paths (list): a list of yaml settings file paths to load
+        settings (dict): optional dictionary of settings values
+
+    Return:
+        dict: settings file containing all settings parsed from the given settings file(s)
+    """
+
+    if settings is None:
+        settings = collections.OrderedDict()
+
+    # add generic global options
+    settings["HOME"] = os.path.expanduser("~")
+    settings["TIMESTAMP"] = time.strftime("%Y%m%d_%H%M%S")
+    settings["HOST_MACHINE_IP"] = _get_ip_address()
+
+    # process settings_file_paths
+    for settings_path in settings_file_paths:
+        yaml_string = _parse_jinja_template(settings_path, template_variables=settings)
+
+        try:
+            settings_from_this_file = yaml.load(yaml_string)
```

Comment:
Codacy found an issue: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=6418257)